internal/gitinterface/keys_test.go

```diff
@@ -0,0 +1,189 @@
+	// if err := repo.SetConfig(&config.Config{
+	// 	Raw: &format.Config{
+	// 		Sections: format.Sections{
+	// 			&format.Section{
+	// 				Name: "user",
+	// 				Options: format.Options{
+	// 					&format.Option{
+	// 						Key:   "foo",
+	// 						Value: "bar",
+	// 					},
+	// 				},
+	// 			},
+	// 		},
+	// 	},
+	// }); err != nil {
+	// 	t.Error(err)
+	// }
+	// _, _, _, err = GetSigningInfo(repo)
+	// assert.ErrorIs(t, err, ErrSigningKeyNotSpecified)
+
+	// This config is set to use ssh.
+	if err := repo.SetConfig(&config.Config{
+		Raw: &format.Config{
+			Sections: format.Sections{
+				&format.Section{
+					Name: "user",
+					Options: format.Options{
+						&format.Option{
+							Key:   "signingkey",
+							Value: "abcdef",
+						},
+					},
+				},
+				&format.Section{
+					Name: "gpg",
+					Options: format.Options{
+						&format.Option{
+							Key:   "format",
+							Value: "ssh",
+						},
+					},
+				},
+			},
+		},
+	}); err != nil {
+		t.Error(err)
+	}
+
+	signingMethod, keyInfo, _, err = getSigningInfo(repo)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, SigningMethodSSH, signingMethod)
+	assert.Equal(t, "abcdef", keyInfo)
+
+	// This config is set to use x509.
+	if err := repo.SetConfig(&config.Config{
+		Raw: &format.Config{
+			Sections: format.Sections{
+				&format.Section{
+					Name: "user",
+					Options: format.Options{
+						&format.Option{
+							Key:   "signingkey",
+							Value: "abcdef",
+						},
+					},
+				},
+				&format.Section{
+					Name: "gpg",
+					Options: format.Options{
+						&format.Option{
+							Key:   "format",
+							Value: "x509",
+						},
+					},
+				},
+			},
+		},
+	}); err != nil {
+		t.Error(err)
+	}
+
+	signingMethod, keyInfo, _, err = getSigningInfo(repo)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, SigningMethodX509, signingMethod)
+	assert.Equal(t, "abcdef", keyInfo)
+
+	// FIXME: We have to mock the system / global configs, this test will fail
+	// because it'll detect the user's configuration.
+	// // This config is set to use some unknown signing method.
+	// if err := repo.SetConfig(&config.Config{
+	// 	Raw: &format.Config{
+	// 		Sections: format.Sections{
+	// 			&format.Section{
+	// 				Name: "user",
+	// 				Options: format.Options{
+	// 					&format.Option{
+	// 						Key:   "signingkey",
+	// 						Value: "abcdef",
+	// 					},
+	// 				},
+	// 			},
+	// 			&format.Section{
+	// 				Name: "gpg",
+	// 				Options: format.Options{
+	// 					&format.Option{
+	// 						Key:   "format",
+	// 						Value: "abcdef",
+	// 					},
+	// 				},
+	// 			},
+	// 		},
+	// 	},
+	// }); err != nil {
+	// 	t.Error(err)
+	// }
+
+	// _, _, _, err = GetSigningInfo(repo)
+	// assert.ErrorIs(t, err, ErrUnknownSigningMethod)
+}
+
+func createTestRepository() (string, error) {
+	testDir, err := os.MkdirTemp("", "gittuf")
+	if err != nil {
+		return testDir, err
+	}
+	cmd := exec.Command("git", "init", testDir)
+	return testDir, cmd.Run()
+}
```

Comment:
I cloned the tests where one set run on the filesystem, the other in memory. The in-memory ones don't seem to work (locally, they pulled up my global config instead of what I was setting for the in memory repo). I'll try to debug this later. https://github.com/adityasaky/gittuf/actions/runs/4200712261/jobs/7287009661#step:4:31

Reply:
Ah of course, this is because we're now shelling out for git config. I'll open a ticket for future reference as a nice to have and mark it resolved here.

Reply:
https://github.com/adityasaky/gittuf/issues/19